src/Sprites/ArtFile.h

```diff
@@ -43,7 +46,17 @@ struct ArtFile
 
 
 	void ValidateImageMetadata() const;
-	void CountFrames(std::size_t& frameCount, std::size_t& subframeCount, std::size_t& unknownCount) const;
+	void CountFrames(std::size_t& frameCount, std::size_t& layerCount, std::size_t& unknownCount) const;
 
 	static const std::array<char, 4> TagPalette;
+
+	struct LayerMetadata {
+		uint8_t layerCount : 7;
+		uint8_t bReadOptionalData : 1;
+	};
+
+	struct UnknownBitfield {
+		uint8_t unknown : 7;
+		uint8_t bReadOptionalData : 1;
+	};
```

Comment:
`LayerMetadata` and `UnknownBitfield` structs can probably be collapsed into a single struct. They have the same layout. I think they might really be two separate instances of the same struct.

Reply:
It is hard for me to say sense we don't know enough about the structs. I think I would prefer leaving them as separate structs until we know more. Otherwise we will have 2 layer counts if we just go with 2 LayerMetadatas.

Reply:
Ahh, but I'm suggesting dropping the "Layer" part of the name in the struct, and make that part of the variable instance name.

That leaves open the question of what do we name the struct. I don't have a suggestion there:
```cpp
struct ThisStructNeedsABetterName {
	uint8_t count : 7;
	uint8_t bHasOptionalData : 1;
};
```